node/src/components/consensus/era_supervisor.rs

```diff
@@ -358,19 +358,36 @@ where
             ))
         }
 
-        let era = Era::new(
+        let mut era = Era::new(
             consensus,
             start_time,
             start_height,
-            newly_slashed,
-            slashed,
+            new_faulty,
+            faulty,
             validators,
         );
+
+        // Mark validators as faulty for which we have evidence in a recent era.
+        for e_id in self.iter_past_other(era_id, self.bonded_eras()) {
+            if let Some(old_era) = self.active_eras.get_mut(&e_id) {
+                for pub_key in old_era.consensus.validators_with_evidence() {
+                    let proposed_blocks = era.resolve_evidence(&pub_key);
```

Comment:
Resolve evidence is side-effecting, so we might want to change its name to `resolve_evidence_and_mark_faulty`

Reply:
Done in https://github.com/casper-network/casper-node/pull/1607/commits/2d39fd4d382e3790f88fdde206a359ef9f94a204.